waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -618,12 +618,15 @@ proc queryLoop(w: WakuStore, query: HistoryQuery, candidateList: seq[RemotePeerI
   var futureList: seq[Future[MessagesResult]]
   for peer in candidateList.items:
     futureList.add(w.queryFromWithPaging(query, peer))
-  futureList = await allFinished(futureList) # TODO: is there a proc that returns a Future[seq[T]?
-  var messagesList: seq[WakuMessage]
-  # all futures have already been awaited, so the following loop runs without actually waiting
-  for fut in futureList:
-    let successResult = await fut
-    if successResult.isOk: messagesList.insert(successResult.value)
+  await allFutures(futureList) # all(), which returns a Future[seq[T]], has been deprecated
+
+  let messagesList = futureList
+    .filterIt(it.completed()) # just a sanity check. These futures have been awaited before using allFutures()
+    .mapIt(it.read())
+    .filterIt(it.isOk)
+    .mapIt(it.value)
+    .concat()
```

Comment:
I like how explicit and readable this is, but it may be worth considering that each `mapIt` and `filterIt` returns a new `seq` which for very large sequences could be inefficient. Not really a problem here, as the `futureList` will be small.
The way to fix that would be to call [a `map` proc](https://nim-lang.org/docs/sequtils.html#map%2CopenArray%5BT%5D%2Cproc%28T%29) only once, but with a slightly more advanced logic. Something like:
```
proc mapToWakuMessages(in: Future[MessagesResult]): seq[WakuMessages] =
  if in.completed() and in.read().isOk():
    in.read().value
  else:
    @[]

let messagesList = futureList.map(mapToWakuMessages).concat()
```
(Might be some errors here, but the rough idea is to minimise seq copies)